File: src/main/java/org/kite9/diagram/style/GridContainerPosition.java
package org.kite9.diagram.style;

public class GridContainerPosition implements ContainerPosition {

	private final IntegerRange x;

	private final IntegerRange y;
	
	public GridContainerPosition(IntegerRange x, IntegerRange y) {
		super();
		this.x = x;
		this.y = y;
	}
	
	public IntegerRange getX() {
		return x;
	}

	public IntegerRange getY() {
		return y;
	}

}

File: a/src/myMain/kotlin/org/kite9/diagram/model/style/GridContainerPosition.kt
package org.kite9.diagram.model.style

import org.kite9.diagram.common.range.IntegerRange

data class GridContainerPosition(val x: IntegerRange, val y: IntegerRange) : ContainerPosition {

    fun isSet(): Boolean {
        return !IntegerRange.notSet(x) && !IntegerRange.notSet(y)
    }

    override fun toString(): String {
        return if (!isSet()) {
            "null"
        } else {
            "[" + x.from + "," + x.to + "," + y.from + "," + y.to + "]"
        }
    }
}